experimental/Schemes/elliptic_surface.jl

```diff
@@ -0,0 +1,1116 @@
+export elliptic_surface, trivial_lattice, weierstrass_model, weierstrass_chart, algebraic_lattice, zero_section, section, relatively_minimal_model, fiber_components, generic_fiber, reducible_fibers, fibration_type, mordell_weil_lattice
+
+@doc raw"""
+    EllipticSurface{BaseField<:Field, BaseCurveFieldType} <: AbsCoveredScheme{BaseField}
+
+The type of a relatively minimal elliptic surface.
+
+A genus $1$-fibration is a proper map
+$\pi \colon X \to C$ to a curve $C$ whose fibers are curves of
+(arithmetic) genus $1$.
+The fibration is relatively minimal if its fibers do not contain any ``(-1)``-curves.
+We call the fibration elliptic if it comes equipped with a section.
+This turns the generic fiber of $\pi$ into an elliptic curve $E/k(C)$ where
+$k(C)$ is the function field of the curve $C$.
+"""
+@attributes mutable struct EllipticSurface{BaseField<:Field, BaseCurveFieldType} <: AbsCoveredScheme{BaseField}
+  Y::CoveredScheme{BaseField}
+  E::EllCrv{BaseCurveFieldType}
+  MWL::Vector{EllCrvPt{BaseCurveFieldType}}
+  MWLtors::Vector{EllCrvPt{BaseCurveFieldType}}
+  Weierstrasschart::AbsSpec
+  Weierstrassmodel::CoveredScheme
+  inc_Weierstrass
+  inc_Y
```

Comment:
```suggestion
  inc_Weierstrass::CoveredClosedEmbedding
  inc_Y::CoveredClosedEmbedding
```
...and probably similar below.